### PR TITLE
Fix remaining clippy warnings

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: src/protocol.rs [hunk 1]
```diff
@@ -23,11 +23,11 @@ pub enum Response {
     /// Response for `Request::Compile`.
     Compile(CompileResponse),
     /// Response for `Request::GetStats`, containing server statistics.
-    Stats(ServerInfo),
+    Stats(Box<ServerInfo>),
```
Comment (froydnj) [inline src/protocol.rs]: Is this (and the `DistClientState` changes related to `Box`) due to `large_enum_variant`?

Comment (ry) [reply src/protocol.rs]: yes

**Comment by ry** [comment]
Ok I think this is ready for review now. It seems to be clippy clean for `--all-targets` and `--no-default-features`.

**Comment by ry** [comment]
@froydnj This PR should be ready to land (the two commits can be squashed). I think it improves the code base and allows us to check clippy for new errors when making new changes. PTAL.

**Comment by ry** [comment]
squashed and updated for Rust 1.42.0 ...

**Pull request merged**